src/electron/adapters/action-message-dispatcher-adapter.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { TelemetryData } from 'common/extension-telemetry-events';
+import { Message } from 'common/message';
+import { Dispatcher } from 'common/message-creators/types/dispatcher';
+
+export class ActionMessageDispatcherAdapter implements Dispatcher {
```

Comment:
it may be better to convey where it is dispatching to instead of just naming it adapter, now that we have an interface.
having ActionMessageDispatcherAdapter  & ActionMessageDispatcher as class names are confusing.

Reply:
@devaradhanm any suggestions? I did not spend much time thinking of the name, I was expecting some feedback about it to be honest

Reply:
Really, both types of dispatcher are dispatching action messages, they're just doing it differently. Maybe the interface can be `ActionMessageDispatcher` and the implementations can be `RemoteActionMessageDispatcher` and `DirectActionMessageDispatcher`?

Reply:
I though about using `Direct`... I'll go with you're suggestion